### PR TITLE
Extra tests

### PR DESCRIPTION
**Pull request by tyll**
These are the tests from #25 that need further review.

### CODE REVIEW

File: tests/ctl/nmstatectl_test.py [hunk 1]
```diff
@@ -16,19 +16,59 @@
...
+@mock.patch('sys.argv', ['nmstatectl', 'show'])
+@mock.patch.object(nmstatectl.netinfo, 'show', lambda: {})
+def test_run_ctl_directly_show_empty():
+    nmstatectl.main()
+
+
+@mock.patch('sys.argv', ['nmstatectl', 'show', '--only', 'eth1'])
+@mock.patch.object(
+    nmstatectl.netinfo, 'show', lambda: json.loads(LO_JSON_STATE))
+@mock.patch('sys.stdout', new_callable=StringIO)
```
[the remark this answers is not in the document]
Comment (tyll) [reply tests/ctl/nmstatectl_test.py]: Using `nmstatectl.sys.stdout` fails:

```
_______________________________________________________________________________ test_run_ctl_directly_show_only_empty _______________________________________________________________________________

thing = <module 'nmstatectl' from '/home/till/scm/nmstate/.tox/check-py36/lib/python3.6/site-packages/nmstatectl/__init__.py'>, comp = 'sys', import_path = 'nmstatectl.sys'

    def _dot_lookup(thing, comp, import_path):
        try:
>           return getattr(thing, comp)
E           AttributeError: module 'nmstatectl' has no attribute 'sys'

/usr/lib64/python3.6/unittest/mock.py:1071: AttributeError

During handling of the above exception, another exception occurred:

args = (), keywargs = {}, extra_args = [], entered_patchers = []
exc_info = (<class 'ModuleNotFoundError'>, ModuleNotFoundError("No module named 'nmstatectl.sys'",), <traceback object at 0x7fd999e0bdc8>)
patching = <unittest.mock._patch object at 0x7fd999d97748>

    @wraps(func)
    def patched(*args, **keywargs):
        extra_args = []
        entered_patchers = []
    
        exc_info = tuple()
        try:
            for patching in patched.patchings:
>               arg = patching.__enter__()

/usr/lib64/python3.6/unittest/mock.py:1171: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/usr/lib64/python3.6/unittest/mock.py:1227: in __enter__
    self.target = self.getter()
/usr/lib64/python3.6/unittest/mock.py:1397: in <lambda>
    getter = lambda: _importer(target)
/usr/lib64/python3.6/unittest/mock.py:1084: in _importer
    thing = _dot_lookup(thing, comp, import_path)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

thing = <module 'nmstatectl' from '/home/till/scm/nmstate/.tox/check-py36/lib/python3.6/site-packages/nmstatectl/__init__.py'>, comp = 'sys', import_path = 'nmstatectl.sys'

    def _dot_lookup(thing, comp, import_path):
        try:
            return getattr(thing, comp)
        except AttributeError:
>           __import__(import_path)
E           ModuleNotFoundError: No module named 'nmstatectl.sys'

/usr/lib64/python3.6/unittest/mock.py:1073: ModuleNotFoundError
__________________________________________________________________________________ test_run_ctl_directly_show_only __________________________________________________________________________________

thing = <module 'nmstatectl' from '/home/till/scm/nmstate/.tox/check-py36/lib/python3.6/site-packages/nmstatectl/__init__.py'>, comp = 'sys', import_path = 'nmstatectl.sys'

    def _dot_lookup(thing, comp, import_path):
        try:
>           return getattr(thing, comp)
E           AttributeError: module 'nmstatectl' has no attribute 'sys'

/usr/lib64/python3.6/unittest/mock.py:1071: AttributeError

During handling of the above exception, another exception occurred:

args = (), keywargs = {}, extra_args = [], entered_patchers = []
exc_info = (<class 'ModuleNotFoundError'>, ModuleNotFoundError("No module named 'nmstatectl.sys'",), <traceback object at 0x7fd999d17e48>)
patching = <unittest.mock._patch object at 0x7fd999d97b70>

    @wraps(func)
    def patched(*args, **keywargs):
        extra_args = []
        entered_patchers = []
    
        exc_info = tuple()
        try:
            for patching in patched.patchings:
>               arg = patching.__enter__()

/usr/lib64/python3.6/unittest/mock.py:1171: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/usr/lib64/python3.6/unittest/mock.py:1227: in __enter__
    self.target = self.getter()
/usr/lib64/python3.6/unittest/mock.py:1397: in <lambda>
    getter = lambda: _importer(target)
/usr/lib64/python3.6/unittest/mock.py:1084: in _importer
    thing = _dot_lookup(thing, comp, import_path)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

thing = <module 'nmstatectl' from '/home/till/scm/nmstate/.tox/check-py36/lib/python3.6/site-packages/nmstatectl/__init__.py'>, comp = 'sys', import_path = 'nmstatectl.sys'

    def _dot_lookup(thing, comp, import_path):
        try:
            return getattr(thing, comp)
        except AttributeError:
>           __import__(import_path)
E           ModuleNotFoundError: No module named 'nmstatectl.sys'

/usr/lib64/python3.6/unittest/mock.py:1073: ModuleNotFoundError


```

File: tests/ctl/compat.py [hunk 2]
```diff
@@ -17,8 +17,23 @@
...
+import sys
+
 try:
     from unittest import mock
 except ImportError:  # py2
     import mock
+
+try:
+    # the python2 io.StringIO() class does not work here
+    from StringIO import StringIO
```
[the remark this answers is not in the document]
Comment (tyll) [reply tests/ctl/compat.py]: done

File: tests/ctl/nmstatectl_test.py [hunk 3]
```diff
@@ -16,19 +16,59 @@
...
+}
+"""
+
 
-@mock.patch('sys.argv', ['nmstatectl', 'show'])
-@mock.patch.object(nmstatectl.netinfo, 'show', lambda: {})
 @mock.patch('sys.argv', ['nmstatectl', 'set', 'mystate.json'])
 @mock.patch.object(nmstatectl.netapplier, 'apply', lambda state: None)
-def test_run_ctl_directly():
+@mock.patch(BUILTINS + '.open', mock.mock_open(read_data='{}'))
```
[the remark this answers is not in the document]
Comment (tyll) [reply tests/ctl/nmstatectl_test.py]: done

File: tests/ctl/nmstatectl_test.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (EdDev) [reply tests/ctl/nmstatectl_test.py]: You used @mock.patch.object(nmstatectl.sys, 'stdout', ..) ?
If you do not use the 'object', then you need to do probably something like:
@mock.patch('nmstatectl.nmstatectl.sys.stdout', ..)

[the remark this answers is not in the document]
Comment (tyll) [reply tests/ctl/nmstatectl_test.py]: thank you, this did the trick

File: tests/ctl/compat.py [hunk 4]
```diff
@@ -17,8 +17,10 @@
 from __future__ import absolute_import
 from __future__ import division
 
+
```
Comment (EdDev) [inline tests/ctl/compat.py]: nit: Unrelated changes (also on line 25).
